### PR TITLE
Documentation fixes with some linting

### PR DESCRIPTION
**Pull request by Darrenmeehan**
*Description of changes:*

- Fixed documentation in `README.md` to work with existing code.
- Update awscli commands to work with version 2.
- Ensure files built when following `README.md` are ignored from `git`
- Remove Travis button as it's no longer used.
- Resolves https://github.com/awslabs/aws-lambda-rust-runtime/issues/84

Some code changes. I can revert this to make this PR cleaner if deemed more appropriate.

- Fixed minor typos in docstring.
- Minor linting.

By submitting this pull request

- [x] I confirm that my contribution is made under the terms of the Apache 2.0 license.
- [x] I confirm that I've made a best effort attempt to update all relevant documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -87,6 +85,7 @@ You can now test the function using the AWS CLI or the AWS Lambda console
 
 ```bash
 $ aws lambda invoke --function-name rustTest \
+  --cli-binary-format raw-in-base64-out \
```
Comment (softprops) [inline README.md:88]: It might be worth calling out you need the aws  v2 cli for this to work. Some are using that but as it's relatively new, many are not.

File: lambda/Cargo.toml [hunk 2]
```diff
@@ -2,6 +2,7 @@
 name = "lambda"
 version = "0.1.0"
 authors = ["David Barsky <dbarsky@amazon.com>"]
```
Comment (softprops) [inline lambda/Cargo.toml:4]: I might drop the "client" part of this. That's an implementation detail most users won't likely care or care to know about. Just calling this "AWS Lambda Rust runtime" might get the point across and be a little bit more inline with https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

File: README.md (hunk 1 above)
Comment (softprops) [reply README.md:88]: For clearer context https://docs.aws.amazon.com/cli/latest/userguide/cliv2-migration.html#cliv2-migration-binaryparam

Comment (Darrenmeehan) [reply README.md:88]: > It might be worth calling out you need the aws v2 cli for this to work. Some are using that but as it's relatively new, many are not.

Great call. Should I add the invokcation command for both versions? Or highlight the need for `--cli-binary-format` with v2? I'm learning towards the latter as it's less verbose.

File: lambda/Cargo.toml (hunk 2 above)
Comment (Darrenmeehan) [reply lambda/Cargo.toml:4]: Thanks for the feedback. I've addresses this in f3337dbd35194865f8d476fa4cf3448265864dc4

File: README.md (hunk 1 above)
Comment (Darrenmeehan) [reply README.md]: I've went with the following but happy to change! 56a745f0c9b87b9dc2ef369846f60a771da0bc17

Comment (davidbarsky) [reply README.md]: Yeah, let’s prefer version two of the CLI. I’d be surprised if the audience for this crate aren’t also early adopters of the newer CLI.

Comment (Darrenmeehan) [reply README.md]: My apologies it was late when I made the above commit https://github.com/awslabs/aws-lambda-rust-runtime/commit/56a745f0c9b87b9dc2ef369846f60a771da0bc17. It should say `most users are using v1`. 

I think as it stands the `README.md` does a good job of documenting the command for both v1 & v2. Is there any change you would like here @davidbarsky ?

Comment (davidbarsky) [reply README.md]: Oh, yeah. No change then. Thanks for doing this!

File: lambda/src/lib.rs [hunk 3]
```diff
@@ -6,10 +6,10 @@
 //! There are two mechanisms of defining a Lambda function:
 //! 1. The `#[lambda]` attribute, which generates the boilerplate needed to
 //!    to launch and run a Lambda function. The `#[lambda]` attribute _must_
-//!    be placed on an asynchronous main funtion. However, asynchronous main
-//!    funtions are not legal valid Rust, which means that a crate like
+//!    be placed on an asynchronous main function. However, asynchronous main
+//!    functions are not legal valid Rust, which means that a crate like
 //!    [Runtime](https://github.com/rustasync/runtime) must be used. A main function
-//!    decorated using `#[lamdba]`
+//!    decorated using `#[lambdas]`
```
Comment (davidbarsky) [inline lambda/src/lib.rs:12]:
```suggestion
//!    decorated using `#[lambda]`s
```

Comment (Darrenmeehan) [reply lambda/src/lib.rs]: Thanks for the suggestion.

I love this new GitHub feature :+1:

File: lambda/src/lib.rs [hunk 4]
```diff
@@ -6,10 +6,10 @@
 //! There are two mechanisms of defining a Lambda function:
 //! 1. The `#[lambda]` attribute, which generates the boilerplate needed to
 //!    to launch and run a Lambda function. The `#[lambda]` attribute _must_
-//!    be placed on an asynchronous main funtion. However, asynchronous main
-//!    funtions are not legal valid Rust, which means that a crate like
+//!    be placed on an asynchronous main function. However, asynchronous main
+//!    functions are not legal valid Rust, which means that a crate like
 //!    [Runtime](https://github.com/rustasync/runtime) must be used. A main function
```
Comment (davidbarsky) [inline lambda/src/lib.rs:11]: (Sorry for the back and forth on this; I reviewed this before my coffee hit.)

Can we reword this to say something along the lines of "Tokio is required; we suggest using `#[tokio::main]`"?

Comment (davidbarsky) [reply lambda/src/lib.rs:11]: Thanks! I only noticed this docs issue now since I wrote that; the two big issues are:

1. `runtime` is deprecated.
2. This runtime has a hard Tokio dependency (for now). Cross-runtime support is too hard, and until Rusoto supports runtimes other than Tokio, I don't think its worth _attempting_ to support other runtimes in here.

**Comment by Darrenmeehan** [comment]
@davidbarsky Could this PR be linked with the following issue please? https://github.com/awslabs/aws-lambda-rust-runtime/issues/84

**Comment by davidbarsky** [comment]
Yup! If you’re to add a resolves [issue number] to the description of this PR, it’ll link and close that issue.

File: lambda/src/lib.rs (hunk 4 above)
Comment (Darrenmeehan) [reply lambda/src/lib.rs]: I'm not totally familiar with `tokio` but I've attempted to clear this section up. Anything you would like changed? Sorry if I'm misunderstanding something!

**Comment by Darrenmeehan** [comment]
Hi @davidbarsky is there anything you would like changed?

**Comment by davidbarsky** [comment]
@Darrenmeehan No, this looks great! Sorry for the delay and thanks for the fix!

**Comment by Darrenmeehan** [comment]
Awesome thank you for the quick feedback. Looking forward to contributing some more.
